### PR TITLE
Refactor contract structure parsing

### PR DESCRIPTION
**Pull request by DavidKnott**
### - What I did
Refactor how smart contracts are parsed.
Fixed a typo in folder naming.
### - How I did it
Broke `parse_tree_to_lll` out into multiple functions
### - How to verify it
Make sure that the functionality remains unchanged 
### - Description for the changelog
None
### - Cute Animal Picture
image

### CODE REVIEW

File: tests/parser/integration/test_crowdfund.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+#     c.participate(value=3, sender=t.k5)
+#     c.participate(value=4, sender=t.k6)
+#     s.head_state.timestamp += 1000
+#     assert c.expired()
+#     assert not c.reached()
+#     pre_bals = [s.head_state.get_balance(x) for x in [t.a3, t.a4, t.a5, t.a6]]
+#     c.refund()
+#     post_bals = [s.head_state.get_balance(x) for x in [t.a3, t.a4, t.a5, t.a6]]
+#     assert [y - x for x, y in zip(pre_bals, post_bals)] == [1, 2, 3, 4]
+
```
Comment (jacqueswww) [inline tests/parser/integration/test_crowdfund.py]: Comment all of this back in ?

File: viper/parser/parser.py [hunk 2]
```diff
@@ -282,55 +282,66 @@ def mk_full_signature(code):
...
+        _contract_defs = _contracts[_contractname]
+        _defnames = [_def.name for _def in _contract_defs]
+        contract = {}
+        if len(set(_defnames)) < len(_contract_defs):
+            raise VariableDeclarationException("Duplicate function name: %s" % [name for name in _defnames if _defnames.count(name) > 1][0])
+        for _def in _contract_defs:
+            sig = FunctionSignature.from_definition(_def)
+            contract[sig.name] = sig
+        external_contracts[_contractname] = contract
+    return external_contracts
```
Comment (jacqueswww) [inline viper/parser/parser.py]: I like it. Just on thing: _contracts can probably be renamed to simple contracts now right?

Comment (DavidKnott) [reply viper/parser/parser.py]: Do you mean changing `_contracts` to `contract?

Comment (jacqueswww) [reply viper/parser/parser.py]: @DavidKnott nope I was just talking about the _contracts part, but now that I read it fully there are few variables we can reme the `_` prefix from :smile: 
```python
def parse_external_contracts(external_contracts, contracts):
    for contract_name in contracts:
        contract_defs = contracts[contract_name]
        _defnames = [_def.name for _def in contract_defs]
        contract = {}
        if len(set(_defnames)) < len(contract_defs):
            raise VariableDeclarationException("Duplicate function name: %s" % [name for name in _defnames if _defnames.count(name) > 1][0])
        for _def in contract_defs:
            sig = FunctionSignature.from_definition(_def)
            contract[sig.name] = sig
        external_contracts[contract_name] = contract
    return external_contracts
```

**Pull request merged**